packages/tc-schema-sdk/data-accessors/relationship-type.js

```diff
@@ -0,0 +1,129 @@
+const TreecreeperUserError = require('../lib/biz-ops-error');
+const {
+	assignMetaProperties,
+	transformPrimitiveTypes,
+} = require('../lib/property-assign');
+
+const BIZ_OPS = 'biz-ops';
+
+const transformRichRelationship = (
+	rootType,
+	property,
+	{ name, from, to, relationship, properties },
+) => {
+	const { direction } = property;
+	let relationshipTo;
+	let hasMany;
+	let dest;
+	if (direction !== 'to' && from.type === rootType) {
+		dest = 'outgoing';
+		relationshipTo = to.type;
+		hasMany = !!to.hasMany;
+	} else {
+		dest = 'incoming';
+		relationshipTo = from.type;
+		hasMany = !!from.hasMany;
+	}
+	return {
+		...property,
+		type: relationshipTo,
+		from: from.type,
+		to: to.type,
+		direction: dest,
+		properties: { ...properties },
+		name,
+		relationship,
+		hasMany,
+	};
+};
+
+const getFromTo = (direction, rootType, otherType) =>
+	direction === 'outgoing' ? [rootType, otherType] : [otherType, rootType];
+
+const getFromRawData = (rootType, propertyName, rawData) => {
```

Comment:
I'm just about able to follow what's going on here. I think this file could benefit from being split into more, smaller functions and variables with longer names to make it clear what the logical steps are e.g.

```
let richRelationshipDefinition = getUserDefiniedRichRelationship(...)

if (!richRelationshipDefinition) {
   richRelationshipDefinition = constructRichRelationshipDefinition(...)
}
```